onnxruntime/python/dlpack/dlpack_converter.cc

```diff
@@ -193,16 +193,17 @@ bool IsContiguousTensor(const DLTensor& tensor) {
 
 }  // namespace
 
-// This function returns a shared_ptr to memory managed DLpack tensor
-// constructed out of OrtValue.
-DLManagedTensor* OrtValueToDlpack(const OrtValue& ort_value) {
+// This function returns a pointer to DLManagedTensor constructed from an OrtValue
+// The OrtValue inside OrtDLManagedTensor will increase its own buffer's ref count by one
+// When the consumer of DLManagedTensor is done with the tensor, it should invoke the deleter.
+DLManagedTensor* OrtValueToDlpack(OrtValue& ort_value) {
   ORT_ENFORCE(ort_value.IsTensor(), "Only tensor type OrtValues are supported");
   OrtDLManagedTensor* ort_dlmanaged_tensor(new OrtDLManagedTensor);
```

Comment:
>new OrtDLManagedTensor [](start = 43, length = 22)

see [https://github.com/microsoft/onnxruntime/pull/6973#discussion_r599848843](https://github.com/microsoft/onnxruntime/pull/6973#discussion_r599848843)

Reply:
this function is actually a mirror of pytorch's implementation  https://github.com/pytorch/pytorch/blob/1a2d3b6a750cc4c1709b64412619b28c737e1a79/aten/src/ATen/DLConvertor.cpp#L187

Reply:
IIUC, the deleter defined in DLManagedTensor is a protocol how itself should be deleted 
So the deleter always expects a raw pointer of the DLManagedTensor type 
To me, it seems a redundant step to wrap DLManagedTensor with a share_ptr or unique_ptr 

Also, this seems to be the standard way of how DLpack should be used 

See mxnet: https://github.com/apache/incubator-mxnet/blob/aa92db02201799841bf52db5a0b8507719f04e61/src/ndarray/ndarray.cc#L453 
tensorflow: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/eager/dlpack.cc#L271